### ISSUE TITLE
receiving CORS error [BUG]

### ISSUE THREAD

**Issue by asimiqbl3** [FujiwaraChoki/MoneyPrinter#141]
![image](https://github.com/FujiwaraChoki/MoneyPrinter/assets/105530591/07312604-7c76-4744-adbe-8dd1ac09c121)

I used your described method of using docker image here https://github.com/FujiwaraChoki/MoneyPrinter/issues/140. and I can see containers are running without any issue. 

![image](https://github.com/FujiwaraChoki/MoneyPrinter/assets/105530591/1d609c32-00df-4871-8ac6-503525e54b45)
here's the backend log 

![image](https://github.com/FujiwaraChoki/MoneyPrinter/assets/105530591/9bc4628d-0b2e-430b-93c5-b40dd237850e)

Environment:
Docker version 24.0.4, build 3713ee1

Running on linux host: 
Linux XXXXX 4.18.0-477.15.1.el8_8.x86_64 #1 SMP Fri Jun 2 08:27:19 EDT 2023 x86_64 x86_64 x86_64 GNU/Linux

This is a node on same network I'm accessing it from a laptop with 192.168.1.202 IP . the Linux has IP 192.168.1.21 . do you think it needs to be accessed from same host i.e., 192.168.1.21 GUI ?

Any kind of help is highly appreciated.

Thanks

**Comment by FujiwaraChoki** [comment]
Please visit localhost:3000 not {IP}:3000

**Comment by vo1d-null** [comment]
Try to start your browser with CORS disabled. For me was "C:\Program Files\Google\Chrome\Application\chrome.exe" --disable-web-security --user-data-dir=~\chromeTemp" . I think there was an extension for firefox that does the same.

**Comment by itbane** [comment]
What fixed this for me is that the frontend is fixed on localhost:8080 -> has to be changed to your backend server IP address/hostname.
  `sed -i 's/localhost/<ipaddress>/' index.html`

**Comment by Chiroyce1** [comment]
Is there any reason why the frontend is separate? Couldn't it be served from the same server using Flask's built-in `render_template` method?

**Comment by pur3coff33** [comment]
did u ran backend/main.py?

cd backend
python main.py

**Comment by asimiqbl3** [comment]
> What fixed this for me is that the frontend is fixed on localhost:8080 -> has to be changed to your backend server IP address/hostname. `sed -i 's/localhost/<ipaddress>/' index.html`

This solution worked for me. it starts generating the video but I'm getting a different error message.  I think it's related to some font not properly configured in imagemagick configure file. I'm trying to figure that out now.

Thanks

**Comment by asimiqbl3** [comment]
> Is there any reason why the frontend is separate? Couldn't it be served from the same server using Flask's built-in `render_template` method?

I've docker running on a separate instance with some beefy specs cpu. I'd like to run it on that. It's a headless without gui so wondering if that's possible.

**Comment by asimiqbl3** [comment]
Well. I found the solution for my problem. Sharing it for other people who may be experiencing similar issue. I had to reinstall 
I needed to compile imagemagick with PNG support. There is a project on github that handels all the steps for us, it's ubuntu/debian only atm.


https://github.com/SoftCreatR/imei/

after installing it took around 20-25 mins to complete the imagemagick install and fixed everything for me. so far so good.

Thanks
